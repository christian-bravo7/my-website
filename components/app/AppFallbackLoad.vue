<template>
  <div
    class="fixed inset-0 z-50 flex justify-center items-center bg-white text-pink-800 dark:bg-blue-600 dark:text-blue-500 text-12xl"
  >
    <SiteLogo class="fill-current pulse-effect" />
  </div>
</template>

<script lang="ts">
import { Vue, Component } from 'nuxt-property-decorator';

import SiteLogo from '~/assets/img/logo.svg?inline';

@Component({
  components: {
    SiteLogo,
  },
})
export default class AppFallbackLoad extends Vue {}
</script>

<style lang="scss" scoped>
.pulse-effect {
  animation-name: pulse;
  animation-duration: 500ms;
  animation-timing-function: ease-in-out;
  animation-iteration-count: infinite;
  animation-direction: alternate;
}

@keyframes pulse {
  0% {
    transform: scale(1);
    opacity: 1;
  }

  100% {
    transform: scale(0.9);
    opacity: 0.5;
  }
}
</style>
