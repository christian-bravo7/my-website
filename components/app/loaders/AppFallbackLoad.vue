<template>
  <div
    class="fixed | inset-0 z-50 | flex justify-center items-center | primary-text bg-white dark:bg-dark-theme-800 "
  >
    <div class="w-64 h-64 sm:w-40 sm:h-40">
      <SiteLogo class="w-full h-full | pulse-effect" />
    </div>
  </div>
</template>

<script lang="ts">
import { Vue, Component } from 'nuxt-property-decorator';

import SiteLogo from '~/assets/img/logo.svg?inline';

@Component({
  components: {
    SiteLogo,
  },
})
export default class AppFallbackLoad extends Vue {}
</script>

<style lang="scss" scoped>
.pulse-effect {
  animation-name: pulse;
  animation-duration: 500ms;
  animation-timing-function: ease-in-out;
  animation-iteration-count: infinite;
  animation-direction: alternate;
}

@keyframes pulse {
  0% {
    transform: scale(1);
    opacity: 1;
  }

  100% {
    transform: scale(0.9);
    opacity: 0.5;
  }
}
</style>
