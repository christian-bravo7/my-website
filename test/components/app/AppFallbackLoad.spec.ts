import AppFallbackLoad from '@/components/app/AppFallbackLoad.vue';
import { mount } from '@vue/test-utils';

describe('AppFallbackLoad component', () => {
  it('should render correclty', () => {
    const wrapper = mount(AppFallbackLoad);

    expect(wrapper.html()).toMatchSnapshot();
  });
});
