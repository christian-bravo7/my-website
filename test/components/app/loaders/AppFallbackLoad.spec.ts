import { mount } from '@vue/test-utils';
import AppFallbackLoad from '~/components/app/loaders/AppFallbackLoad.vue';

describe('AppFallbackLoad component', () => {
  it('should render correclty', () => {
    const wrapper = mount(AppFallbackLoad);

    expect(wrapper.html()).toMatchSnapshot();
  });
});
