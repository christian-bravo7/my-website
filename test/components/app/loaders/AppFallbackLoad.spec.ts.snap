// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppFallbackLoad component should render correclty 1`] = `
<div class="fixed | inset-0 z-50 | flex justify-center items-center | primary-text bg-white dark:bg-dark-theme-800">
  <div class="w-64 h-64 sm:w-40 sm:h-40"><svg class="w-full h-full | pulse-effect"></svg></div>
</div>
`;
